openwisp_monitoring/monitoring/static/monitoring/js/chart-utils.js

```diff
@@ -1,22 +1,61 @@
 'use strict';
-const timeRangeKey = 'ow2-chart-time-range';
+const isCustomDateRange = 'ow2-chart-custom-daterange'; // true/false
+const timeRangeKey = 'ow2-chart-time-range'; // 30d
+const startDayKey = 'ow2-chart-start-day'; // September 3, 2022
+const endDayKey = 'ow2-chart-end-day'; // October 3, 2022
+const startDateTimeKey = 'ow2-chart-start-datetime'; // 2022-09-03 00:00:00
+const endDateTimeKey = 'ow2-chart-end-datetime'; // 2022-09-03 00:00:00
 
 django.jQuery(function ($) {
   $(document).ready(function () {
+    var pickerStart, pickerEnd, pickerDays, pickerChosenLabel, start = moment(), end = moment();
+      function initDateRangePickerWidget(start, end) {
+        $('#daterangepicker-widget span').html(start.format('MMMM D, YYYY') + ' - ' + end.format('MMMM D, YYYY'));
+        $("[data-range-key='1 day']").attr('data-time', '1d');
+        $("[data-range-key='3 days']").attr('data-time', '3d');
+        $("[data-range-key='1 week']").attr('data-time', '7d');
+        $("[data-range-key='1 month']").attr('data-time', '30d');
+        $("[data-range-key='1 year']").attr('data-time', '365d');
+        $("[data-range-key='Custom Range']").attr('data-time', 'Custom Range');
+      }
+
+      $('#daterangepicker-widget').daterangepicker({
+        startDate: start,
+        endDate: end,
+        maxDate: moment(),
+        maxSpan: {
+          "year": 1,
+        },
+        ranges: {
+          '1 day': [moment().subtract(1, 'days'), moment()],
+          '3 days': [moment().subtract(3, 'days'), moment()],
+          '1 week': [moment().subtract(7, 'days'), moment()],
+          '1 month': [moment().subtract(30, 'days'), moment()],
+          '1 year': [moment().subtract(365, 'days'), moment()],
+        }
+      }, initDateRangePickerWidget);
+      initDateRangePickerWidget(start, end);
```

Comment:
Why are calling this function from here? Does the `daterangepicker` not call the callback function?

Reply:
When we select default time ranges and reload the page, then the `data-time` attribute won't be set and the charts will not be visible to users if we remove this call.
I'll try to update this logic :+1: